### PR TITLE
[query] Use buffered aggregator to lower TableKeyByAndAgg

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
Should work now...

**Comment by tpoterba** [comment]
@patrick-schultz assigned this to you but it doesn't really need a full review -- I already reviewed the original PR, but made this one to add the fix.

File: hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala [hunk 1]
```diff
@@ -803,52 +807,69 @@ object LowerTableIR {
       case TableKeyByAndAggregate(child, expr, newKey, nPartitions, bufferSize) =>
```
Comment (patrick-schultz) [inline hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala:807]: Could you fix or delete the todo on the above line? (Not sure if it still ignores nPartitions)

**Review by patrick-schultz — CHANGES_REQUESTED** [review]
Ah hah!

File: hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala [hunk 2]
```diff
@@ -820,12 +820,19 @@ object LowerTableIR {
...
+
         val shuffled = ctx.backend.lowerDistributedSort(
           ctx, partiallyAggregated , sortFields, relationalLetsAbove, withNewKeyRType)
         val repartitioned = shuffled.repartitionNoShuffle(shuffled.partitioner.strictify)
 
-        val takeAggSig = PhysicalAggSig(Take(), TakeStateSig(VirtualTypeWithReq(shuffledRowType ,TypeWithRequiredness(shuffledRowType))))
-        val aggStateSigsPlusTake = aggs.states ++ Array(TakeStateSig(VirtualTypeWithReq(shuffledRowType ,TypeWithRequiredness(shuffledRowType))))
+        val takeVirtualSig = TakeStateSig(VirtualTypeWithReq(shuffledRowType ,TypeWithRequiredness(shuffledRowType)))
+        val takeAggSig = PhysicalAggSig(Take(), takeVirtualSig)
+        val aggStateSigsPlusTake = aggs.states ++ Array(takeVirtualSig)
```
Comment (patrick-schultz) [inline hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala:835]: Also, this looks unused.

Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala:835]: used on L872

Comment (patrick-schultz) [reply hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala:835]: Oh sorry, misread the diff and thought you added it on the last commit.

**Review by patrick-schultz — APPROVED** [review]

**Comment by tpoterba** [comment]
This should pass now.

**Pull request merged**